### PR TITLE
8253765: C2: Control randomization in StressLCM and StressGCM

### PR DESCRIPTION
**Pull request by robcasloz**
Use the compilation-local seed in `StressLCM` and `StressGCM` rather than the global one. As a consequence, these options use by default a fresh seed in every compilation, unless `StressSeed=N` is specified, in which case they behave deterministically. Annotate tests that use `StressLCM` and `StressGCM` with the `stress` and `randomness` keys to reflect this change in default behavior.

Tested on `tier1` and on all test cases that use `StressLCM` and `StressGCM` (10 times each).

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- |
| Build | ⏳ (3/3 running) | ⏳ (2/2 running) | ⏳ (2/2 running) |

### Issue
 * [JDK-8253765](https://bugs.openjdk.java.net/browse/JDK-8253765): C2: Control randomization in StressLCM and StressGCM 

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**) ⚠️ Review applies to 64c653881727d41000cdc2231e0446f31d5c7653
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**) ⚠️ Review applies to 64c653881727d41000cdc2231e0446f31d5c7653

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/572/head:pull/572`
`$ git checkout pull/572`

### CODE REVIEW

**Comment by robcasloz** [comment]
/summary
Use the compilation-local seed in 'StressLCM' and 'StressGCM' rather than the
global one. As a consequence, these options use by default a fresh seed in every
compilation, unless 'StressSeed=N' is specified, in which case they behave
deterministically. Annotate tests that use 'StressLCM' and 'StressGCM' with the
'stress' and 'randomness' keys to reflect this change in default behavior.

**Review by vnkozlov — APPROVED** [review]
Good.

**Comment by robcasloz** [comment]
> Good.

Thanks for reviewing Vladimir!

File: src/hotspot/share/opto/c2_globals.hpp [hunk 1]
```diff
@@ -53,7 +53,7 @@
           "Randomize worklist traversal in IGVN")                           \
                                                                             \
   product(uint, StressSeed, 0, DIAGNOSTIC,                                  \
-          "Seed for IGVN stress testing (if unset, a random one is "        \
+          "Seed for randomized stress testing (if unset, a random one is "  \
```
Comment (TobiHartmann) [inline src/hotspot/share/opto/c2_globals.hpp:56]: The description string misses a `)`

**Review by TobiHartmann — APPROVED** [review]
Otherwise looks good.

File: src/hotspot/share/opto/c2_globals.hpp (hunk 1 above)
Comment (robcasloz) [reply src/hotspot/share/opto/c2_globals.hpp:56]: Thanks, will fix!

**Pull request closed without merging**